### PR TITLE
feat: create debug actions

### PR DESCRIPTION
**Pull request by im-coder-lg**
Edit the `build.yml` file to work for release and use `debug-builds.yaml` for instant files. Obviously for testing purposes, like packaging tests, etc.

### CODE REVIEW

**Comment by im-coder-lg** [comment]
https://github.com/Futura-Py/weather/blob/ce8306141bb491da970139cf692451d4103ed060/.github/workflows/build.yml#LL3C9-L3C9

This is an issue, since we want these files only to be released as part of a release workflow. @Moosems any idea?

**Comment by im-coder-lg** [comment]
Got one. Using workflow dispatch, we can put the release version number(semantic) and push for release.

**Comment by Moosems** [comment]
> [`on: push`](https://github.com/Futura-Py/weather/blob/ce8306141bb491da970139cf692451d4103ed060/.github/workflows/build.yml#LL3C9-L3C9)
> From `build.yaml`
> This is an issue, since we want these files only to be released as part of a release workflow. @Moosems any idea?

On release....

**Comment by Moosems** [comment]
I would instead just make a workflow and releases called `Official Releases` instead if you want solid ones for pushes to master.

**Comment by im-coder-lg** [comment]
I think we can use workflow_dispatch, use the variables in it and create the release maybe?

**Comment by Moosems** [comment]
?

**Comment by im-coder-lg** [comment]
We use the workflow_dispatch feature, input some variables(i.e. release version) before running the script, and when the builds are ready, the release is created but drafted, so that we can edit the release changelog, etc.

**Comment by sumeshir26** [comment]
> 

We did a similar thing back during TimerX: [link](https://github.com/Futura-Py/TimerX/blob/master/.github/workflows/release-all.yml)

**Comment by Moosems** [comment]
You can just make it run on release instead

**Comment by Moosems** [comment]
I guess I'm confused what the goal is here

**Comment by im-coder-lg** [comment]
> You can just make it run on release instead

Yeah, we can. But it is better to enable a workflow that does the same. We can then hit the auto-changelog generation button, and voila!

**Comment by Moosems** [comment]
I guess my issue is I'm confused about what the point of it is.

**Comment by Moosems** [comment]
Automation and simplification of what? The builds are already automated and to get zips just open the branch, click code, and download zip.

**Comment by im-coder-lg** [comment]
No, not that way. Using the workflows, we can enable it on demand, i. e. on release. It will draft the release, and all we have to do is edit it to put the right release name, tag and changelog. So, it is much more easier than before, right?

**Comment by im-coder-lg** [comment]
https://github.com/Futura-Py/weather/assets/76845820/0d223f9d-49ab-4881-9ff2-de30a25bc42b

Preview of the debug build workflow. All artifacts are zipped so you have to unzip before running them. This ran on my local OS, Zorin 16.2 with the base OS as Ubuntu 20.04. [The `glibc` issue I told about](https://github.com/Futura-Py/weather/issues/10) can be fixed this way, so it is backwards-compatible, plus it works the same way on Jammy Jellyfish. Anyone having a computer powerful enough to test for Fedora, Arch and the like?

**Comment by im-coder-lg** [comment]
No, it's fixed. You just needed Ubuntu 20.04 since it has GLIBC 2.31. GLIBC 2.35 is luckily backwards-compatible with GLIBC 2.31, so it's done. However, you can discuss with the PyInstaller people to figure out a way to change GLIBC locks, which I _guess_ is hard to do.

File: .github/workflows/debug-builds.yaml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        mkdir Weather
+        mv ./dist/Weather.app ./Weather/Weather.app
+        ln -s /Applications ./Weather/Applications
+        hdiutil create -srcfolder ./Weather Weather.dmg
+        
+    - name: Upload a Build Artifact
+      uses: actions/upload-artifact@v3.1.2
+      with:
+        # Artifact name
+        name: Weather-Mac
```
Comment (im-coder-lg) [inline .github/workflows/debug-builds.yaml:50]: Now, if anybody is wondering, the artifacts are zipped, always. It doesn't show the extension though. Protocol, I guess. Same for the other two steps for the other two OSes.

**Comment by im-coder-lg** [comment]
I got just one more thing to do for this PR. Make the build file now run on demand, just like TimerX.

**Comment by im-coder-lg** [comment]
Alright, ready for review, if you exclude the fact that there are less runners for the workflow to run so it WILL fail for now.

**Comment by rdbende** [comment]
But why not simply create a workflow that runs when a repo release is published, and pull the version number from the `setup.py` or `pyproject.toml`?

````yaml
on:
  release:
    types: [published]

**Comment by Moosems** [comment]
Thats what I had recommended.

**Pull request merged**